### DISCUSSION TITLE
Could this be a good fit for a browser game?

### DISCUSSION THREAD

**Discussion by bliitzkrieg** [supabase/supabase#751] [Questions]
Hey!

I am building an [opensource turn based game](https://github.com/Saturns-Gift/Vision) and I'm in the planning phase just fleshing out my technology stack and I think Supabase is a good fit but I have some questions if I could get some information.

1. Where can I find information on how functions will work? I am thinking about running the game loop on a 30 second tick for each turn. I was thinking of utilizing a 30 second CRON to trigger my game loop and digest my players actions. Each player has 30 seconds to commit actions in the game simultaneously and then it gets batched processed and the state is broadcasted out to the players which will trigger animations on their client.

2. Is it possible to subscribe to a real-time query that has a where clause? Saturns Gift will have a world map where each player will be moving around travelling across the world and I want to limit what the player can see like a fog of war. If this is not possible, is it something that may come in the future?

Thanks and hopefully this all makes sense.

**Comment by kiwicopple** [comment]
Hey @bliitzkrieg. Your game sounds awesome. Hopefully we can be the first to trial it if you use Supabase to build it :)

Caveat these answers with the fact that I have never actually built a game before:

> Where can I find information on how functions will work?

We haven't published much yet just becasue we are more focused on Storage. Functions will probably come in Q2

> I was thinking of utilizing a 30 second CRON to trigger my game loop and digest my players actions. 

This will defintiely be achievable with the functionality we have planned. Essentially the functions will act as a "cron" too if you need it.

> subscribe to a real-time query that has a where clause

Yup! https://supabase.io/docs/client/subscribe#listening-to-row-level-changes

> I want to limit what the player can see like a fog of war

It sounds like it might be better to fetch new state from the database? How will you know what to "send" from the realtime server? Or will you be continuously blasting out everything that's happening on the map?
